app/scripts/migrations/160.ts

```diff
@@ -0,0 +1,314 @@
+      ),
+    );
+
+    return state;
+  }
+
+  if (
+    !hasProperty(networkControllerState, 'networkConfigurationsByChainId') ||
+    !isObject(networkControllerState.networkConfigurationsByChainId)
+  ) {
+    global.sentry?.captureException?.(
+      new Error(
+        'Skipping migration: `NetworkController.networkConfigurationsByChainId` is not an object.',
+      ),
+    );
+
+    return state;
+  }
+
+  if (!permissionControllerState || !isObject(permissionControllerState)) {
+    global.sentry?.captureException?.(
+      new Error(
+        'Skipping migration: `PermissionController` state not found or is not an object.',
+      ),
+    );
+
+    return state;
+  }
+
+  if (!isObject(permissionControllerState.subjects)) {
+    global.sentry?.captureException?.(
+      new Error(
+        'Skipping migration: `PermissionController.subjects` state is not an object.',
+      ),
+    );
+
+    return state;
+  }
+
+  if (
+    !selectedNetworkControllerState ||
+    !isObject(selectedNetworkControllerState)
+  ) {
+    global.sentry?.captureException?.(
+      new Error(
+        'Skipping migration: `SelectedNetworkController` state not found or is not an object.',
+      ),
+    );
+
+    return state;
+  }
+
+  if (!isObject(selectedNetworkControllerState.domains)) {
+    global.sentry?.captureException?.(
+      new Error(
+        'Skipping migration: `SelectedNetworkController.domains` state is not an object.',
+      ),
+    );
+
+    return state;
+  }
+
+  const currentChainId = getChainId(
+    networkControllerState.selectedNetworkClientId,
+    networkControllerState.networkConfigurationsByChainId,
+  );
+
+  if (typeof currentChainId !== 'string') {
+    global.sentry?.captureException?.(
+      new Error(
+        'Skipping migration: No currently selected chainId was found, or the chainId resolved from the `NetworkController.selectedNetworkClientId` value is not a string.',
+      ),
+    );
+
+    return state;
+  }
+
+  const updatedSubjects: string[] = [];
+
+  // Add permission to use the current globally selected network to all Snaps
+  // that have the `endowment:ethereum-provider` permission.
+  const entries = Object.entries(permissionControllerState.subjects) as [
+    string,
+    GenericPermissionControllerSubject,
+  ][];
+
+  permissionControllerState.subjects = entries.reduce<
+    PermissionControllerSubjects<PermissionConstraint>
+  >((accumulator, [key, subject]) => {
+    const permissionKeys = Object.keys(subject.permissions);
+    const needsMigration = permissionKeys.includes(
+      SnapEndowments.EthereumProvider,
+    );
+
+    if (!needsMigration) {
+      return {
+        ...accumulator,
+        [key]: subject,
+      };
+    }
+
+    updatedSubjects.push(key);
+
+    const caip25PermissionCaveatWithCurrentChainIdSet = addPermittedEthChainId(
+      getExistingCaip25PermissionCaveat(subject),
+      currentChainId,
+    );
```

Comment:
we may want to do a `getEthAccounts` and setEthAccounts` to ensure the existing accounts are set on all scopes? @adonesky1

Reply:
Not exactly sure what you mean. Happy to address this in a follow up though!